cot/src/form/fields/chrono.rs

```diff
@@ -118,11 +128,636 @@ impl SelectChoice for Weekday {
+
+        // https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/input/datetime-local#setting_timezones
+        hidden_tag
+            .attr("name", format!("__{}_timezone", self.id()))
+            .attr("id", format!("__{}_timezone", self.id()));
+
+        let mut dt_tag = HtmlTag::input("datetime-local");
+        dt_tag.attr("name", self.id());
+        dt_tag.attr("id", self.id());
+
+        if self.options.required {
+            dt_tag.bool_attr("required");
+        }
+        if let Some(max) = self.custom_options.max {
+            dt_tag.attr(
+                "max",
+                max.naive_local()
+                    .format(BROWSER_DATETIME_WITHOUT_SEC_FMT)
+                    .to_string(),
+            );
+        }
+        if let Some(min) = self.custom_options.min {
+            dt_tag.attr(
+                "min",
+                min.naive_local()
+                    .format(BROWSER_DATETIME_WITHOUT_SEC_FMT)
+                    .to_string(),
+            );
+        }
+
+        if let Some(readonly) = self.custom_options.readonly {
+            if readonly {
+                dt_tag.bool_attr("readonly");
+            }
+        }
+
+        if let Some(value) = &self.value {
+            dt_tag.attr("value", value);
+            let tz = DateTime::parse_from_str(value, "%Y-%m-%dT%H:%M %z")
+                .expect("invalid datetime")
+                .timezone();
+            hidden_tag.attr("value", tz.to_string());
+        }
+
+        if let Some(step) = &self.custom_options.step {
+            let step_value = match step {
+                Step::Any => "any".to_string(),
+                Step::Value(v) => v.num_seconds().to_string(),
+            };
+            dt_tag.attr("step", step_value);
+        }
+        let tag = Html::new(format!("{}{}", hidden_tag.render(), dt_tag.render()));
+
+        write!(f, "{}", tag.as_str())
+    }
+}
+
+impl AsFormField for DateTime<FixedOffset> {
+    type Type = DateTimeWithTimezoneField;
+
+    fn clean_value(field: &Self::Type) -> Result<Self, FormFieldValidationError>
+    where
+        Self: Sized,
+    {
+        let value = check_required(field)?;
+        // Browsers only support naive datetime.
+        let naive = parse_datetime_with_fallback(value)?;
+        // default to UTC if offset(timezone) is not provided.
+        let tz = field.custom_options.timezone.unwrap_or(Tz::UTC);
+
+        let date_time = match tz.from_local_datetime(&naive) {
+            LocalResult::Single(dt) => dt,
+            LocalResult::Ambiguous(dt1, dt2) => {
+                if let Some(prefer_latest) = field.custom_options.prefer_latest {
+                    if prefer_latest { dt2 } else { dt1 }
+                } else {
+                    return Err(FormFieldValidationError::from_string(format!(
+                        "Ambiguous local datetime `{naive:?}`. set the `prefer_latest` option to resolve datetime."
```

Comment:
Do you think it is useful to have these errors as `FormFieldValidationError` variants so that users can match on them?

Reply:
How big of a change would this require? Do you mind creating an issue to track this, and if no one gets to it before I do, I'll take a stab at it.

Reply:
I see I didn't express myself clearly last time. What I meant is that `FormFieldValidationError` is a general form field error / specific field type agnostic. Unless I'm misunderstanding you, adding that error as its variant would break that premise. Is it unwanted? I'm not sure, I'm just extra cautious, so it doesn't grow too much. My initial thought was that we would have to introduce field-type-specific errors and somehow incorporate them into the error, which would potentially require a larger refactor. But I suppose we could just keep adding variants and possibly be fine with it. What do you think @m4tx?

Reply:
That makes sense, I'm convinced. @ElijahAhianyo, one last thing, but please add those error variants as you wanted in the beginning!